### PR TITLE
Build desktop-sim and desktop-sim-test with ASAN and UBSAN

### PR DESCRIPTION
**Pull request by mcm001**
Currently the build goes kaboom on our circular buffer tests lmao

### CODE REVIEW

File: desktop_projects/desktop_sim/tests/test_expressions.cpp [hunk 1]
```diff
@@ -9,18 +9,28 @@
...
+class ExpressionStorageTests : public testing::Test {
+ protected:
+  void SetUp() override {
+    hm_hardwareInit();
+    cli_init();
+    cli_setDefaultConfig();
+    expressionStore.init();
+    expressionStore.removeExpressionsForTrigger(0);
+    expressionStore.removeExpressionsForTrigger(1);
+    expressionStore.removeExpressionsForTrigger(2);
```
Comment (mcm001) [inline desktop_projects/desktop_sim/tests/test_expressions.cpp:21]: why are these here? (just document)

Comment (SamHaggans) [reply desktop_projects/desktop_sim/tests/test_expressions.cpp:21]: They were in there from when there would be default configurations set by the CLI if it didn't detect any written configuration but that doesn't happen anymore so will remove.

**Review by SamHaggans — APPROVED** [review]

**Comment by mcm001** [comment]
Obama prism
